src/Omega_h_pool_kokkos.hpp

```diff
@@ -0,0 +1,122 @@
+//
+// Created by Matthew McCall on 6/12/23.
+//
+// Derived from
+// https://github.com/matthew-mccall/kokkos-memory-pool/blob/8e0a45a5b5d6823976d867e20d742314b6a1d268/src/MemoryPool/MemoryPool.hpp
+
+#ifndef OMEGA_H_KOKKOS_POOL_HPP
+#define OMEGA_H_KOKKOS_POOL_HPP
+
+#include <cstddef>
+#include <list>
+#include <map>
+#include <set>
+#include <utility>
+
+#include "Kokkos_Core.hpp"
+
+namespace Omega_h {
+
+constexpr size_t DEFAULT_BYTES_PER_CHUNK = 1000;
+
+// Total initial size in bytes will subsequently be DEFAULT_NUMBER_OF_CHUNKS * DEFAULT_BYTES_PER_CHUNK
+constexpr size_t DEFAULT_NUMBER_OF_CHUNKS = 700'000;
+
+// If the largest StaticKokkosPool is 2Kb, then the next StaticKokkosPool will be at least 4kB resulting in a total pool size of 2Kb + 4Kb = 6Kb
+constexpr size_t DEFAULT_GROWTH_CONSTANT = 2;
```

Comment:
@cwsmith if the initial pool size has enough of an effect on compile and run time, something to think about is try and relate it to initial mesh size.
Another thing: to have memory in multiples of 2^X,might be useful

Reply:
runtime vs initial pool size: Matthew's testing on the DeltaWing cases (see https://github.com/matthew-mccall/omega_h-kokkos-mempool-paper/) indicates that the cost to resize when the initial pool size is 'small' is minimal (less than 2-3% it seems) relative to a 'large' (using a-pirori knowledge of memory usage) initial pool.

2^x sizing: Thank you.  That may help.  We are looking at some other approaches and trying to see if there are other opportunities like this.